homeassistant/components/flo/sensor.py

```diff
@@ -0,0 +1,213 @@
+            f"{device.mac_address}_current_system_mode",
+            NAME_CURRENT_SYSTEM_MODE,
+            device,
+        )
+        self._state: str = None
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return the current system mode."""
+        if not self._device.current_system_mode:
+            return STATE_UNKNOWN
+        return self._device.current_system_mode
+
+
+class FloCurrentFlowRateSensor(FloEntity):
+    """Monitors the current water flow rate."""
+
+    def __init__(self, device):
+        """Initialize the flow rate sensor."""
+        super().__init__(
+            f"{device.mac_address}_current_flow_rate", NAME_FLOW_RATE, device
+        )
+        self._state: float = None
+
+    @property
+    def icon(self) -> str:
+        """Return the daily usage icon."""
+        return GAUGE_ICON
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current flow rate."""
+        if self._device.current_flow_rate is None:
+            return None
+        return round(self._device.current_flow_rate, 1)
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return gallons as the unit measurement for water."""
+        return "gpm"
+
+
+class FloTemperatureSensor(FloEntity):
+    """Monitors the temperature."""
+
+    def __init__(self, device):
+        """Initialize the temperature sensor."""
+        super().__init__(f"{device.mac_address}_temperature", NAME_TEMPERATURE, device)
+        self._state: float = None
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current temperature."""
+        if self._device.temperature is None:
+            return None
+        return round(fahrenheit_to_celsius(self._device.temperature), 1)
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return gallons as the unit measurement for water."""
+        return TEMP_CELSIUS
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the device class for this sensor."""
+        return DEVICE_CLASS_TEMPERATURE
+
+
+class FloPressureSensor(FloEntity):
+    """Monitors the water pressure."""
+
+    def __init__(self, device):
+        """Initialize the pressure sensor."""
+        super().__init__(
+            f"{device.mac_address}_water_pressure", NAME_WATER_PRESSURE, device
+        )
+        self._state: float = None
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current water pressure."""
+        if self._device.current_psi is None:
+            return None
+        return round(self._device.current_psi, 1)
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return gallons as the unit measurement for water."""
+        return PRESSURE_PSI
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the device class for this sensor."""
+        return DEVICE_CLASS_PRESSURE
+
+
+class FloWifiStrengthSensor(FloEntity):
```

Comment:
I truly hate these entities. They mean so little and are always spamming. Please disable it by default by having `entity_registry_enabled_default` return `False`

Reply:
I can just remove it. I was adding it because the data was there. If anyone complains in the future we can deal then

Reply:
removed